pkg/p2p/libp2p/libp2p.go

```diff
@@ -93,6 +93,7 @@ type Options struct {
 	LightNodeLimit int
 	WelcomeMessage string
 	Transaction    []byte
+	HostFactory    func(context.Context, ...libp2p.Option) (host.Host, error)
```

Comment:
Somehow, this field does not seems useful for the exposed package api, only for the tests.

Would you consider having an unexported field in `hostFactory func(context.Context, ...libp2p.Option) (host.Host, error)` instead to be set only by a new helper function defined in export_test.go. This is just a suggestion, not a hard requirement for the PR approval.